assets/src/edit-story/migration/migrations/v0015_inlineTextProperties.js

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+function inlineTextProperties({ pages, ...rest }) {
+  return {
+    pages: pages.map(reducePage),
+    ...rest,
+  };
+}
+
+function reducePage({ elements, ...rest }) {
+  return {
+    elements: elements.map(updateElement),
+    ...rest,
+  };
+}
+
+function updateElement(element) {
+  if (element.type === 'text') {
+    return updateTextContent(element);
+  }
+
+  return element;
+}
+
+function updateTextContent({
+  bold,
+  fontWeight,
+  fontStyle,
+  textDecoration,
+  letterSpacing,
+  color,
+  content,
+  ...rest
+}) {
+  // We use an array to chain all the converters more nicely
+  const convertedContent = [content]
+    .map((c) => convertInlineBold(c, bold, fontWeight))
+    .map((c) => convertInlineItalic(c, fontStyle))
+    .map((c) => convertInlineUnderline(c, textDecoration))
+    .map((c) => addInlineColor(c, color))
+    .map((c) => addInlineLetterSpacing(c, letterSpacing))
+    .pop();
+
+  return { ...rest, content: convertedContent };
+}
+
+function convertInlineBold(content, isBold, fontWeight) {
+  // Do we have a specific global weight to apply for entire text field?
+  const globalWeight =
+    typeof fontWeight === 'number' && fontWeight != 400
+      ? fontWeight
+      : isBold === true
+      ? 700
+      : null;
+
+  if (globalWeight) {
+    // In that case, strip any inline bold from the text and wrap everything in a span with correct style
+    const stripped = stripTag(content, 'strong');
+    const fancyBold = `font-weight: ${globalWeight}`;
+    const wrapped = wrapWithSpan(stripped, fancyBold);
+    return wrapped;
+  }
+
+  const justBold = 'font-weight: 700';
```

Comment:
Maybe I'm misreading this, but it looks like `700` gets handled by the `if (globalWeight) {}` above? In either case, I assume `isBold === false` is a valid case here too?

Reply:
There's a difference between the two handlers:

- The global weight in the top wraps the entire text field in a single `<span style="font-weight: X;">` while removing any and all inline `<strong>` tags. Hence `stripTag` and `wrapWithSpan`.
- The inline replacement in the bottom replaces existing `<strong>` tags with a similar inline `<span style="font-weight: X;">`, but not globally - just inline and in multiple places. Hence `replaceTagWithSpan`.

So yes, if `isBold` is not `true` and if `fontWeight` is missing or `400`, we'll just convert any existing `strong` tags with _bolded_ `span`'s - and if there aren't any of those tags, nothing happens.

Reply:
Got it.